PolyEngine/Engine/Src/TimeSystem.cpp

```diff
@@ -0,0 +1,60 @@
+#include "EnginePCH.hpp"
+
+#include "TimeSystem.hpp"
+#include <chrono>
+
+using std::chrono::steady_clock;
+using std::chrono::duration_cast;
+using std::chrono::duration;
+
+//------------------------------------------------------------------------------
+void Poly::TimeSystem::TimeUpdatePhase(World * world)
+{
+	TimeWorldComponent* timeComponent = &world->GetTimeWorldComponent();
+
+	//Get new delta time
+	std::chrono::steady_clock::time_point frameTime = steady_clock::now();
+	duration<float> deltaTime = duration_cast<duration<float>>(frameTime - timeComponent->LastFrameTime);
```

Comment:
Don't know how much do we care, but game time should actually be of `double` type. Float's precision is simply too low. After the game has been running for a day the precision drops to 7.8 milliseconds, and after a week to 62.5 milliseconds! (admittedly that's a pretty rare case) And then you've got stuff like occasional catastrophic cancellations when calculating delta time.